src/test/java/com/google/cloud/compute/v1/it/ITInstanceGroupManagerTest.java

```diff
@@ -0,0 +1,233 @@
+                            .build())
+                    .addNetworkInterfaces(
+                        NetworkInterface.newBuilder().setNetwork("global/networks/default").build())
+                    .build())
+            .build();
+    completedOperation =
+        waitForOperation(
+            instanceTemplateClient.insertInstanceTemplate(
+                ProjectName.of(DEFAULT_PROJECT), instanceTemplate));
+    String instanceTemplateSelfLink = completedOperation.getTargetLink();
+    resourcesToCleanUp.put("instance-template", instanceTemplateSelfLink);
+
+    // Create an instance group manager
+    InstanceGroupManager instanceGroupManager =
+        InstanceGroupManager.newBuilder()
+            .setBaseInstanceName("base-instance-" + ID)
+            .setName("test-instance-group-" + ID)
+            .setTargetSize(0)
+            .setInstanceTemplate(instanceTemplateSelfLink)
+            .build();
+    completedOperation =
+        waitForOperation(
+            instanceGroupManagerClient.insertInstanceGroupManager(
+                PROJECT_ZONE_NAME, instanceGroupManager));
+    instanceGroupManagerSelfLink = completedOperation.getTargetLink();
+    assertNotNull(instanceGroupManagerSelfLink);
+    resourcesToCleanUp.put("instance-group-manager", instanceGroupManagerSelfLink);
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    for (String instanceGroupManager : resourcesToCleanUp.get("instance-group-manager")) {
+      waitForOperation(instanceGroupManagerClient.deleteInstanceGroupManager(instanceGroupManager));
+    }
+    for (String instanceTemplate : resourcesToCleanUp.get("instance-template")) {
+      waitForOperation(instanceTemplateClient.deleteInstanceTemplate(instanceTemplate));
+    }
+    for (String disk : resourcesToCleanUp.get("disk")) {
+      waitForOperation(diskClient.deleteDisk(disk));
+    }
+
+    diskClient.close();
+    instanceGroupManagerClient.close();
+    instanceTemplateClient.close();
+    globalOperationClient.close();
+    zoneOperationClient.close();
+  }
+
+  @Test
+  public void getInstanceGroupManagerTest() {
+    InstanceGroupManager instanceGroupManager =
+        instanceGroupManagerClient.getInstanceGroupManager(instanceGroupManagerSelfLink);
+    assertEquals(instanceGroupManagerSelfLink, instanceGroupManager.getSelfLink());
+  }
+
+  @Test
+  public void listInstanceGroupManagerTest() {
+    // list instance groups and ensure we find the one we just created
+    List<InstanceGroupManager> instanceGroupManagers =
+        Lists.newArrayList(
+            instanceGroupManagerClient.listInstanceGroupManagers(PROJECT_ZONE_NAME).iterateAll());
+    assertThat(instanceGroupManagers).isNotNull();
+    assertThat(instanceGroupManagers.size()).isGreaterThan(0);
+    boolean found = false;
+    for (InstanceGroupManager manager : instanceGroupManagers) {
+      if (manager.getSelfLink().equals(instanceGroupManagerSelfLink)) {
+        found = true;
```

Comment:
just return here...

Reply:
Good point, in a previous iteration, these were in the same test so we wanted to continue. Will fix.